src/persist-client/src/operators/shard_source.rs

```diff
@@ -316,6 +325,16 @@ where
                         let bytes_emitted = {
                             let mut bytes_emitted = 0;
                             for part_desc in std::mem::take(&mut batch_parts) {
+                                // TODO(mfp): Push the filter down into the Subscribe?
+                                if cfg.dynamic.stats_filter_enabled() {
+                                    let should_fetch = part_desc.stats.as_ref().map_or(true, |stats| !filter.should_fetch(stats));
+                                    if should_fetch {
```

Comment:
I think this is backwards 😱

Reply:
welp that would certainly explain some of the correctness issues :D